src/Aspire.Dashboard/Components/Pages/Resources.razor.cs

```diff
@@ -294,6 +296,84 @@ private async Task ExecuteResourceCommandAsync(ResourceViewModel resource, Comma
         }
     }
 
+    private static string? GetSourceTooltip(ResourceViewModel resource)
+    {
+        if (resource.IsProject() && resource.TryGetProjectPath(out var projectPath))
+        {
+            return Path.GetFileName(projectPath);
+        }
+
+        if (resource.TryGetExecutablePath(out var executablePath))
+        {
+            resource.TryGetExecutableArguments(out var arguments);
+            var argumentsString = arguments.IsDefaultOrEmpty ? "" : string.Join(" ", arguments);
+
+            return $"{Path.GetFileName(executablePath)} {argumentsString}";
+        }
+
+        if (resource.TryGetContainerImage(out var containerImage))
+        {
+            return containerImage;
+        }
+
+        if (resource.Properties.TryGetValue(KnownProperties.Resource.Source, out var value) && value.HasStringValue)
+        {
+            return value.StringValue;
+        }
+
+        return null;
```

Comment:
Please de-duplicate this with the code in `SourceColumnDisplay`. It'll be easy for these to fall out of sync with each other.

Doing this should improve the code in both places. It can be done with a new static method that returns a few different values:

- display value (for column)
- full value (for copy/tooltip
- sub text (optional, for `ContentAfterValue`)